client/my-sites/media-library/test/list.jsx

```diff
@@ -3,209 +3,287 @@
+			render( <MediaList { ...props } /> );
+
+			await user.click( screen.getByTestId( `list-item-${ media[ 0 ].ID }` ) );
+			expect( selectMediaItems ).toHaveBeenNthCalledWith( 1, DUMMY_SITE_ID, [ media[ 0 ] ] );
 
-		test( 'allows selecting a single item', () => {
-			toggleItem( 0 );
-			expectSelectedItems( 0 );
-			toggleItem( 2 );
-			expectSelectedItems( 2 );
+			await user.click( screen.getByTestId( `list-item-${ media[ 2 ].ID }` ) );
+			expect( selectMediaItems ).toHaveBeenNthCalledWith( 2, DUMMY_SITE_ID, [ media[ 2 ] ] );
 		} );
 
-		test( 'allows deselecting a single item', () => {
-			toggleItem( 0 );
-			expectSelectedItems( 0 );
-			toggleItem( 0 );
-			expectSelectedItems();
+		test( 'allows deselecting a single item', async () => {
+			const user = userEvent.setup();
+			const WrappedMediaList = withSelectedItems( MediaList );
+			render( <WrappedMediaList { ...props } /> );
+
+			await user.click( screen.getByTestId( `list-item-${ media[ 0 ].ID }` ) );
+			expect( selectMediaItems ).toHaveBeenNthCalledWith( 1, DUMMY_SITE_ID, [ media[ 0 ] ] );
+
+			await user.click( screen.getByTestId( `list-item-${ media[ 0 ].ID }` ) );
+			expect( selectMediaItems ).toHaveBeenNthCalledWith( 2, DUMMY_SITE_ID, [] );
 		} );
 
-		test( 'only selects a single item when selecting multiple via Shift+click', () => {
-			toggleItem( 0 );
-			toggleItem( 4, true ); // Shift+click to select items 0 through 4
-			expectSelectedItems( 4 );
+		test( 'only selects a single item when selecting multiple via Shift+click', async () => {
+			const user = userEvent.setup();
+			render( <MediaList { ...props } /> );
+
+			await user.click( screen.getByTestId( `list-item-${ media[ 0 ].ID }` ) );
+			expect( selectMediaItems ).toHaveBeenNthCalledWith( 1, DUMMY_SITE_ID, [ media[ 0 ] ] );
+
+			await user.keyboard( '{Shift>}' );
+			await user.click( screen.getByTestId( `list-item-${ media[ 4 ].ID }` ) );
+			expect( selectMediaItems ).toHaveBeenNthCalledWith( 2, DUMMY_SITE_ID, [ media[ 4 ] ] );
 		} );
 	} );
 
 	describe( 'ungrouped sources', () => {
-		const getList = ( media, source ) => {
-			return mount(
-				<MediaList
-					filterRequiresUpgrade={ false }
-					site={ { ID: DUMMY_SITE_ID } }
-					media={ media }
-					mediaScale={ 0.24 }
-					moment={ moment }
-					source={ source }
-					single
-					selectedItems={ [] }
-					selectMediaItems={ selectMediaItems }
-				/>
-			)
-				.find( MediaList )
-				.instance();
+		const props = {
+			filterRequiresUpgrade: false,
+			media,
+			mediaScale: 0.24,
+			moment,
+			selectMediaItems,
+			selectedItems: [],
+			single: true,
+			site: { ID: DUMMY_SITE_ID },
 		};
 
 		test( 'should have no group label for an ungrouped source', () => {
-			const grid = getList( fixtures.media, 'pexels' ).render();
-			expect( grid.props.getGroupLabel() ).toEqual( '' );
+			render( <MediaList { ...props } source="pexels" /> );
+			expect( screen.queryAllByText( /September/ ) ).toEqual( [] );
```

Comment:
Isn't it a bit more accurate to assert that:

```suggestion
			expect( screen.queryAllByText( /September/ ) ).toHaveLength( 0 );
```